httprouter/apirest/errors.go

```diff
@@ -0,0 +1,129 @@
+	ErrVochainGetTxFailed                                      // vochain GetTx failed
+	ErrVochainReturnedErrorCode                                // vochain replied with error code
+	ErrVochainReturnedInvalidElectionID                        // vochain returned an invalid electionID after executing tx
+	ErrVochainReturnedWrongMetadataCID                         // vochain returned an unexpected metadata CID after executing tx
+	ErrMarshalingServerJSONFailed                              // marshaling (server-side) JSON failed
+	ErrCantFetchElectionList                                   // cannot fetch election list
+	ErrCantFetchElection                                       // cannot fetch election
+	ErrCantFetchElectionResults                                // cannot fetch election results
+	ErrCantFetchTokenTransfers                                 // cannot fetch token transfers
+	ErrCantFetchEnvelopeHeight                                 // cannot fetch envelope height
+	ErrCantFetchEnvelope                                       // cannot fetch vote envelope
+	ErrCantCheckTxType                                         // cannot check transaction type
+	ErrCantABIEncodeResults                                    // cannot abi.encode results
+	ErrCantComputeKeyHash                                      // cannot compute key hash
+	ErrCantAddKeyAndValueToTree                                // cannot add key and value to tree
+	ErrCantAddKeyToTree                                        // cannot add key to tree
+	ErrCantGenerateFaucetPkg                                   // cannot generate faucet package
+	ErrCantEstimateBlockHeight                                 // cannot estimate startDate block height
+	ErrCantMarshalMetadata                                     // cannot marshal metadata
+	ErrCantPublishMetadata                                     // cannot publish metadata file
+	ErrTxTypeMismatch                                          // transaction type mismatch
+	ErrElectionIsNil                                           // election is nil
+	ErrElectionResultsNotYetAvailable                          // election results are not yet available
+	ErrElectionResultsIsNil                                    // election results is nil
+	ErrElectionResultsMismatch                                 // election results don't match reported ones
+	ErrMissingModulesForHandler                                // missing modules attached for enabling handler
+	ErrHandlerUnknown                                          // handler unknown
+	ErrHTTPRouterIsNil                                         // httprouter is nil
+	ErrBaseRouteInvalid                                        // base route must start with /
+)
+
+func (e APIError) Error() string {
+	return e.String()
+}
+
+func (e APIError) HTTPStatusCode() int {
+	switch e {
+	case ErrElectionNotFound:
+		return http.StatusNotFound
```

Comment:
here we would add all the cases we want to treat specially (i.e. not return a simple 400, but a 404 or 204)